pinot-server/src/main/java/org/apache/pinot/server/starter/helix/SegmentOnlineOfflineStateModelFactory.java

```diff
@@ -46,6 +50,13 @@
  * 3. Delete an existed segment.
  */
 public class SegmentOnlineOfflineStateModelFactory extends StateModelFactory<StateModel> {
+  // NOTE: Helix might process CONSUMING -> DROPPED transition as 2 separate transitions: CONSUMING -> OFFLINE followed
+  // by OFFLINE -> DROPPED. Use this cache to track the segments that just went through CONSUMING -> OFFLINE transition
+  // to detect CONSUMING -> DROPPED transition.
+  // TODO: Check how Helix handle CONSUMING -> DROPPED transition and remove this cache if it's not needed.
+  private final Cache<Pair<String, String>, Boolean> _recentlyOffloadedConsumingSegments =
```

Comment:
How about move the cache into the tableDataMgr? Considering the key of the map is pair<tableName, segmentName> anyway.

Reply:
I prefer keeping it here since it is just a workaround for Helix state transition handling. Once Helix fixes the problem we should be able to remove it